src/intervals.jl

```diff
@@ -0,0 +1,159 @@
+using .IntervalArithmetic
+
+function evaluate(a::Taylor1, dx::Interval)
+    dx == (-1..1) && return _evaluate(a, dx, Val(true))
+    dx == (0..1) && return _evaluate(a, dx, Val(false))
+    # Usual Horner rule
+    @inbounds begin
+        suma = a[end]*one(dx)
+        for k in a.order-1:-1:0
+            suma = suma*dx + a[k]
+        end
+    end
+    return suma
+end
+
+function _evaluate(a::Taylor1, dx::Interval, ::Val{true})
+    if iseven(a.order)
+        @inbounds begin
+            suma1 = a[end-1]
+            for k in a.order-3:-2:1
+                suma1 += a[k]
+            end
+            suma2 = a[end]
+            for k in a.order-2:-2:2
+                suma2 += a[k]
+            end
+        end
+    else
+        @inbounds begin
+            suma1 = a[end]
+            for k in a.order-2:-2:1
+                suma1 += a[k]
+            end
+            suma2 = a[end-1]
+            for k in a.order-3:-2:2
+                suma2 += a[k]
+            end
+        end
+    end
+    return a[0] + suma1*dx + suma2 * (one(dx) * (0..1))
```

Comment:
I think the last term should be written as Interval{T} (0, 1) for type stability. The one should not be necessary.

Reply:
The `one(dx)` was introduced to assure type stability :-) 

I will change it as suggested, thanks!

Reply:
Implemented in [03e448e](/JuliaDiff/TaylorSeries.jl/pull/195/commits/03e448edd524aedfbfa8dc844621dbccd0d8260d).